gonull.go

```diff
@@ -59,6 +61,8 @@ func (n Nullable[T]) Value() (driver.Value, error) {
 // UnmarshalJSON implements the json.Unmarshaler interface for Nullable, allowing it to be used as a nullable field in JSON operations.
 // This method ensures proper unmarshalling of JSON data into the Nullable value, correctly setting the Valid flag based on the JSON data.
 func (n *Nullable[T]) UnmarshalJSON(data []byte) error {
+	n.Present = true
```

Comment:
I think `n.Present = true` should be added in `Scan` method as well.

```go
func (n *Nullable[T]) Scan(value interface{}) error {
	n.Present = true
...
```

Reply:
And same for `NewNullable` func.

```go
func NewNullable[T any](value T) Nullable[T] {
	return Nullable[T]{Val: value, Valid: true, Present: true}
}
```

Reply:
addressed